File: frontend/src/models/data/VerdicalDivider.ts
import DataDivider from "./DataDivider";

export default class VerdicalDivider extends DataDivider {
    frequency: number;

    constructor() {
        super();
        this.frequency = 1;
    }

    evaluate(x: number, y: number): boolean {
        const theta = this.deg * Math.PI / 180;
        const sinT = Math.sin(theta);
        const cosT = Math.cos(theta); 
        return cosT * (y + this.shiftY) - sinT * (x - this.shiftX) < Math.tan((sinT * (y + this.shiftY) + cosT * (x - this.shiftX)) * this.frequency)*1000 + 20;
    }
}

File: backend/Models/VerdicalDivider.cs
namespace Graphic_Neural_Network.backend.Models
{
    using System;

    public class VerdicalDivider : DataDivider
    {
        public double Frequency { get; set; }
        public override string Type => "verdicaldivider";

        public VerdicalDivider()
        {
            Frequency = 1;
        }

        public override bool Evaluate(double x, double y)
        {
            double theta = Deg * Math.PI / 180;
            double sinT = Math.Sin(theta);
            double cosT = Math.Cos(theta);

            double shiftedY = y + ShiftY;
            double shiftedX = x - ShiftX;

            double projection = sinT * shiftedY + cosT * shiftedX;
            return cosT * shiftedY - sinT * shiftedX < Math.Tan(projection * Frequency) * 1000 + 20;
        }
    }

}
